### PR TITLE
SPARK-33755: Allow creating orc table when row format separator is defined

### PR DESCRIPTION
**Pull request by StefanXiepj**
### What changes were proposed in this pull request?

When creating table like this:
`create table test_orc(c1 string) row format delimited fields terminated by '002' stored as orcfile;`
spark throws exception like :
`Operation not allowed: ROW FORMAT DELIMITED is only compatible with 'textfile', not 'orcfile'(line 2, pos 0)`

In this pr, we support non-strict rules when creating orc table with row format delimited.

### Why are the changes needed?

I found this problem when migrating task from hive to spark. Hive is supported (It's not good, but it's not a problem, we can ignore it). So I fixed it in version Spark 2.4. Although Orc doesn't need this delimiter, but I don't think we need to be so strict in syntax. It is more convenient to migrate tasks from hive to spark.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs to be added

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala [hunk 1]
```diff
@@ -153,4 +153,18 @@ class HiveOrcSourceSuite extends OrcSuite with TestHiveSingleton {
...
   }
+
+  test("SPARK-33755: Allow creating orc table when row format separator is defined") {
+    withTable("row_format_orc") {
+      sql(
+        s"""CREATE TABLE row_format_orc(
+           |  intField INT,
+           |  stringField STRING
+           |)
+           |ROW FORMAT DELIMITED FIELDS TERMINATED BY '002'
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: How does the ORC table wotk with the delimiter?

Comment (StefanXiepj) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: Although this unnecessary for ORC table, we can support an option that user can choose to ignore it or not.

Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: It's better to be explicit for what Spark supports. It's a bit odd that we add no-op syntax.

Comment (StefanXiepj) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: We have many tasks which are running on hive and slowly, so we want to migrate it from hive to spark. I think other companies want to this too. If we support ignore this exception by set spark.sql.orc.skipRowFormatDelimitedError=true , we can migrate hive task and user do not need to modify his sql script.

Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: Does Hive work with this delimiter specified with ORC?

Comment (StefanXiepj) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: Hive doesn't work with it, but hive doesn't throw exception too.

Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: Let's don't fix it then. It's odd that Spark works with a syntax that's no-op, and that does not also work in Hive.

Comment (StefanXiepj) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala:164]: thx very much, i'll close it.

**Pull request closed without merging**